pgcli/pgexecute.py

```diff
@@ -278,8 +278,12 @@ def connect(
         if not self.host:
             self.host = self.get_socket_directory()
 
-        pid = self._select_one(cursor, "select pg_backend_pid()")[0]
-        self.pid = pid
+        if self.dbname == "pgbouncer":
```

Comment:
I think that the better, pythonic way is to `try` the query, then catch an `except`ion if something goes wrong. I don't know this code well so probably here are some downsides I don't see. Potentially this will allow `pgcli` to connect not only to the pgbouncer, but also to any other "limited" databases :)

Reply:
I thought about it as well. But do you have the "invalid command select pg_backend_pid()" error when connecting to databases through `pgbouncer`? Or is it the error only specific to `pgbouncer` admin database? In latter case, I would prefer to make a localized fix, rather than make it too wide and potentially suppress / not be aware of other reasons why this query can fail.

Also, can you post the relevant line from `~/.config/pgcli/log` when the error happens? I'd like to see which exception type is being thrown.

Reply:
The "try/catch" approach does have its upsides. For example, if I create a fake database and name it `pgbouncer` and try to connect to it, my current fix will lie to me about `pid`. :) That's not a good thing. Perhaps I will go with the `try/catch`, but I'd still need to know which exception to catch. I'm not fond of catch-all `except`.

Reply:
Here is my `~/.config/pgcli/log` with relevant errors: https://gist.github.com/and-semakin/ec4c9431fd843ce0ecd918830f31f9bf